internal/state/relationship/capturer.go

```diff
@@ -155,3 +221,31 @@ func getBackendServiceNamesFromRoute(hr *v1beta1.HTTPRoute) map[types.Namespaced
 
 	return svcNames
 }
+
+// matchingGateways looks through all existing label selectors defined by listeners in a gateway,
+// and if any matches are found, returns a map of those gateways
+func (c *CapturerImpl) matchingGateways(labelMap map[string]string) map[types.NamespacedName]struct{} {
+	gateways := make(map[types.NamespacedName]struct{})
+	for gw, selectors := range c.gatewayLabelSelectors {
+		for _, selector := range selectors {
+			if selector.Matches(labels.Set(labelMap)) {
+				gateways[gw] = struct{}{}
+				break
+			}
+		}
+	}
+
+	return gateways
+}
+
+func (c *CapturerImpl) removeGatewayLabelSelector(gatewayName types.NamespacedName) {
+	delete(c.gatewayLabelSelectors, gatewayName)
+	for ns, cfg := range c.namespaces {
+		delete(cfg.gateways, gatewayName)
+		c.namespaces[ns] = cfg
+	}
+}
+
+func (n namespaceCfg) match() bool {
```

Comment:
could we bring it closer to namespaceCfg definition?  so that it is easier to read  and maintain

```
type namespaceCfg struct {
...
}

func (n namespaceCfg) match() bool {
...
}
```